### DISCUSSION TITLE
new ui.table and event rowClick

### DISCUSSION THREAD

**Discussion by bobanovo** [zauberzeug/nicegui#664] [Q&A]
Hi, 
I would like to ask for help. For `ui.table` I need to listen, on which row was clicked, and process the id of row in the rest of the code.
When I used `ui.table` (qtable), and `table.on('rowClick', callback_function)`, the function has only one parameter, which doesn't contain the id of the row, or data of specific row etc. What do I wrong?

```py
tbl1=ui.table(columns=columns_, rows=rows_, row_key='id', selection='single').props('')
print(tbl1.on('rowClick', lambda msg: print("> ",msg)))
```

and the output:
```
>  {'id': 4, 'type': 'rowClick', 'args': {'isTrusted': True, '_vts': 1680074011030}}
>  {'id': 4, 'type': 'rowClick', 'args': {'isTrusted': True, '_vts': 1680074011218}}
>  {'id': 4, 'type': 'rowClick', 'args': {'isTrusted': True, '_vts': 1680074011840}}
>  {'id': 4, 'type': 'rowClick', 'args': {'isTrusted': True, '_vts': 1680074012014}}
```

Thanks a lot, for your help.
Radek

**Comment by bobanovo** [comment]
additional comment - the table has 3rows, and the id in output is still the same, although I clicked on the different rows...
quasar QTable in event has two additional parameters, but here is only one, with {id (the same ids), type and args}, without relevance to clicked row...

in ag-grid, there is cellClicked, and it works for me, but I would like use qtable for data-table (with sorting, search input, and pagination)...

**Comment by falkoschindler** [comment]
That's a good question, @bobanovo!

Currently, there seems to be no way to get the row ID in a rowClick event. This is because NiceGUI's event system only expects one argument to the event handler:
https://github.com/zauberzeug/nicegui/blob/21c22828d98ee0320be4970f34c98849bc444a65/nicegui/templates/index.html#L57

But QTable's rowClick has the signature `(evt, row, index) => void` with the second and third argument being the most interesting. I'm not sure, how to incorporate additional arguments into the event message. It's already pretty complicated with the `args` parameter for filtering individual arguments, but only if `e` is an object... I'll have to think about it.

**Comment by falkoschindler** [comment]
I created issue https://github.com/zauberzeug/nicegui/issues/672 to track the development of this enhancement. Let's continue the discussion over there.
